packages/x-date-pickers/src/models/adapters.ts

```diff
@@ -58,6 +58,11 @@ export interface AdapterFormats {
    * @example "00"
    */
   seconds: string;
+  /**
+   * The seconds without leading zeros.
+   * @example "0"
+   */
+  secondsNoLeadingZeros: string;
```

Comment:
Strictly speaking, people should never override this format 
It's used for technical purposed, not to render any data, so if someone replace it with a 1-based value, the whole component would stop working.

I added here anyway because it's super unlikely that anybody would do such a thing, but I can replace it with a property on the adapter to be safer if you want.

Reply:
Have you considered not adding this format, but hardcoding the `s` usage?
Yes, it's a bit risky - we need to make sure that if we introduce a new adapter that doesn't have such a format, we handle it accordingly. 🤔 
But, it would avoid the potential problem of allowing a change that could break functionality. 🤷

Reply:
> Maybe I could add a test in the gregorian adapter that checks if adapter.formatByString(someDate, 's') does what I want him to do, that would makethe approach less risky.

Sounds like a great solution! 💙 👍